src/main/java/de/tum/in/www1/artemis/service/LectureService.java

```diff
@@ -77,11 +98,40 @@ public Lecture filterActiveAttachments(Lecture lectureWithAttachments, User user
 
     /**
      * Deletes the given lecture.
-     * Attachments are not explicitly deleted, as the delete operation is cascaded by the database.
+     * Attachments and Lecture Units are not explicitly deleted, as the delete operation is cascaded by the database.
      * @param lecture the lecture to be deleted
      */
     public void delete(Lecture lecture) {
-        lectureRepository.delete(lecture);
+        Optional<Lecture> lectureToDeleteOptional = lectureRepository.findByIdWithStudentQuestionsAndLectureUnitsAndLearningGoals(lecture.getId());
```

Comment:
please use Transactional here

Reply:
Done in ae8bec507791f3f824dd5125a53b82621e651ead